packages/backend/app/Http/Controllers/MessagesController.php

```diff
@@ -2,6 +2,96 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Message;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\Routing\ResponseFactory;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Validator;
+use Ramsey\Uuid\Uuid;
+
+const DEFAULT_PER_PAGE = 10;
+
 class MessagesController extends Controller
 {
+    /**
+     * メッセージ投稿用エンドポイント
+     *
+     * @param  Request  $request
+     * @return Response|JsonResponse|Application|ResponseFactory
+     */
+    public function createMessage(Request $request): Response|JsonResponse|Application|ResponseFactory
+    {
+        $validator = Validator::make($request->all(), [
+            'body' => ['required', 'string', 'max:140'],
+        ]);
+
+        if ($validator->fails()) {
+            return response([
+                'message' => 'Validation Failed',
+                'errors' => $validator->errors(),
+            ], 400);
+        }
+
+        if (is_null($request->user())) {
+            return response([
+                'message' => 'Internal Server Error',
+            ], 500);
+        }
+
+        $message = new Message();
+        $message->id = Uuid::uuid7()->toString();
+        $message->user_id = $request->user()->id;
+        $message->body = $validator->getData()['body'];
+        $message->save();
+
+        return response(null, 201);
+    }
+
+    /**
+     * メッセージ取得用エンドポイント
+     *
+     * @param  Request  $request
+     * @return Response|JsonResponse|Application|ResponseFactory
+     */
+    public function listMessage(Request $request): Response|JsonResponse|Application|ResponseFactory
+    {
+        $validator = Validator::make($request->all(), [
+            'lastMessageId' => ['string'],
+            'perPage' => ['integer'],
+        ]);
+
+        if ($validator->fails()) {
+            return response([
+                'message' => 'Validation Failed',
+                'errors' => $validator->errors(),
+            ], 400);
+        }
+
+        if (is_null($request->user())) {
+            return response([
+                'message' => 'Internal Server Error',
+            ], 500);
+        }
+
+        $perPage = $validator->getData()['perPage'] ?? DEFAULT_PER_PAGE;
+
+        $messages = Message::query()
+            ->select(['id', 'user_id', 'body', 'created_at'])
+            ->where('user_id', $request->user()->id)
+            ->when($validator->getData()['lastMessageId'] ?? null, function ($query, $lastMessageId) {
+                return $query->where('id', '<', $lastMessageId);
+            })
```

Comment:
[IMO]
処理内容がシンプルなのであれば無名関数の短縮記法の方がシンプルに書けます。
```suggestion
            ->when(
                $validator->getData()['lastMessageId'] ?? null,
                fn(Builder $query, int $lastMessageId): Builder => $query->where('id', '<', $lastMessageId)
            )
```